### PR TITLE
Set framebuffer layout from EmuWindow.

### PR DESCRIPTION
**Pull request by bunnei**
Precursor PR to touchpad support. This moves the logic for setting the render window framebuffer layout (e.g. width, height, top screen boundaries, bottom screen boundaries) from the renderer to EmuWindow the implementation. Because for now our two EmuWindow implementations use the same basic layout (top screen + bottom screen scaled to the window size), I made a default function for computing the layout part of EmuWindow so that it can be shared. But this design is more flexible and should allow for more creative things in the future (e.g. making the lower screen a thumbnail, or putting it on a separate window).

### CODE REVIEW

**Comment by purpasmart96** [comment]
Yeah this is good, no since duplicating code.

File: src/common/emu_window.h [hunk 1]
```diff
@@ -38,6 +39,17 @@ class EmuWindow
         std::pair<unsigned,unsigned> min_client_area_size;
     };
 
+    /// Describes the layout of the window framebuffer (size and top/bottom screen positions)
+    struct FramebufferLayout {
+        FramebufferLayout() : width(0), height(0), top_screen({}), bottom_screen({}) { }
+        FramebufferLayout(int width, int height);
+
+        int width;
```
Comment (neobrain) [inline src/common/emu_window.h]: should be unsigned

File: src/citra_qt/main.cpp [hunk 2]
```diff
@@ -193,7 +196,10 @@ void GMainWindow::OnDisplayTitleBars(bool show)
 void GMainWindow::BootGame(std::string filename)
 {
     LOG_INFO(Frontend, "Citra starting...\n");
-    System::Init(render_window);
+    System::Init();
+
+    VideoCore::g_renderer->SetWindow(render_window);
```
Comment (neobrain) [inline src/citra_qt/main.cpp]: Why was this moved out of System::Init?

File: src/common/emu_window.h [hunk 3]
```diff
@@ -38,6 +39,17 @@ class EmuWindow
         std::pair<unsigned,unsigned> min_client_area_size;
     };
 
+    /// Describes the layout of the window framebuffer (size and top/bottom screen positions)
+    struct FramebufferLayout {
+        FramebufferLayout() : width(0), height(0), top_screen({}), bottom_screen({}) { }
+        FramebufferLayout(int width, int height);
```
Comment (neobrain) [inline src/common/emu_window.h]: Please make this constructor a factory method instead, i.e. "static FramebufferLayout DefaultScreenLayout(int width, int height)". That way, it's more clear that the frontend is still free to chose its own screen layout.

File: src/common/emu_window.h [hunk 4]
```diff
@@ -59,6 +68,14 @@ class EmuWindow
     static void KeyReleased(KeyMap::HostDeviceKey key);
 
     /**
+     * Factory method for constructing a default FramebufferLayout
+     * @param width Window framebuffer width in pixels
+     * @param height Window framebuffer height in pixels
+     * @return Newly created FramebufferLayout object with the default screen regions initialized
+     */
+    static FramebufferLayout DefaultScreenLayout(int width, int height);
```
Comment (neobrain) [inline src/common/emu_window.h]: Sorry for not being clear, I was meaning to have that static function as part of FramebufferLayout. It doesn't really make a lot of sense to have a factory method as part of the parent class.

**Comment by neobrain** [comment]
Added one minor nitpick, other than that LGTM.
